src/matrix/net/RequestScheduler.ts

```diff
@@ -50,6 +50,13 @@ class Request implements IHomeServerRequest {
     response(): Promise<any> {
         return this._responsePromise;
     }
+
+    async responseCode(): Promise<number> {
+        let resolve;
+        const promise: Promise<number> = new Promise(r => resolve = r);
+        this.requestResult?.responseCode().then(code => resolve(code));
+        return promise;
+    }
```

Comment:
I'm not sure if this is what I'm supposed to do.. wdyt?

Reply:
Let's discuss OOB